WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -745,10 +895,19 @@ private extension EditableOrderViewModel {
             allProductVariations.append(variation)
         }
 
-        let input = OrderSyncProductInput(product: .variation(variation), quantity: 1)
-        orderSynchronizer.setProduct.send(input)
-
-        analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
+        // Single-Selection
+        guard featureFlagService.isFeatureFlagEnabled(.productMultiSelectionM1), isProductMultiSelectionBetaFeatureEnabled else {
+            let input = OrderSyncProductInput(product: .variation(variation), quantity: 1)
+            orderSynchronizer.setProduct.send(input)
+            analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
+            return
+        }
+        // Multi-Selection
+        if !selectedProductVariations.contains(where: { $0?.productVariationID == variation.productVariationID }) {
+            selectedProductVariations.append(variation)
+        } else {
+            selectedProductVariations.removeAll(where: { $0?.productVariationID == variation.productVariationID })
+        }
```

Comment:
Stylistically, it's probably better to use an `if` here. 

A guard reads more like a check, or a precondition – "check x and don't proceed if you're not in the right state". You can tell that's what it's for because of the requirement for an early return.

In this case, we've got two equally valid ways of the function working, and you can even imagine a situation where you might want more common behaviour to happen after the feature flagged branches are run (e.g. analytics, logging, etc.)


```suggestion
        if featureFlagService.isFeatureFlagEnabled(.productMultiSelectionM1), isProductMultiSelectionBetaFeatureEnabled {
            // Multi-Selection
            if !selectedProductVariations.contains(where: { $0?.productVariationID == variation.productVariationID }) {
                selectedProductVariations.append(variation)
            } else {
                selectedProductVariations.removeAll(where: { $0?.productVariationID == variation.productVariationID })
            }
        } else {
            // Single-Selection
            let input = OrderSyncProductInput(product: .variation(variation), quantity: 1)
            orderSynchronizer.setProduct.send(input)
            analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
        }
        
```

Reply:
Oh that's interesting, I was using guard mainly for removing if-else statements, but that makes sense. Changed on 83fb666f8f103b77c45898f1cebfd75f4e4c6a01

Reply:
Yeah... there's nothing actually wrong with an if-else statement, and sometimes, like this, it's the right tool for the job.

You can kind of think of guard as if it's a physical person, guarding the rest of the method from unacceptable input or being run at the wrong time. It's a simplification, but sometimes helpful to choose between them.